### PR TITLE
Add ecs workflow.

### PR DESCRIPTION
**Pull request by terabytesoftw**
| Q             | A
| ------------- | ---
| Is bugfix?    | ❌
| New feature?  | ✔️
| Breaks BC?    | ❌

### CODE REVIEW

**Review by vjik — CHANGES_REQUESTED** [review]
Why need to use ECS?
PHP CS Fixer supports PER CS, why need something else?

File: .github/workflows/ecs.yml [hunk 1]
```diff
@@ -0,0 +1,81 @@
...
+        description: Composer command to run.
+        default: composer update --prefer-dist --no-interaction --no-progress --optimize-autoloader --ansi
+        required: false
+        type: string
+      composer-version:
+        description: Composer version to use.
+        default:
+        required: false
+        type: string
+      coverage:
```
Comment (vjik) [inline .github/workflows/ecs.yml:16]: It's need for CS fix?

File: .github/workflows/ecs.yml [hunk 2]
```diff
@@ -0,0 +1,81 @@
...
+        default:
+        required: false
+        type: string
+      ini-values:
+        description: Initial values for PHP configuration.
+        default: date.timezone='UTC'
+        required: false
+        type: string
+      os:
+        description: Os to test, separated by comma.
```
Comment (vjik) [inline .github/workflows/ecs.yml:32]:
```suggestion
        description: OS to run action, separated by comma.
```

File: .github/workflows/ecs.yml [hunk 3]
```diff
@@ -0,0 +1,81 @@
...
+        required: true
+        type: string
+      php:
+        description: PHP versions to test, separated by comma.
```
Comment (vjik) [inline .github/workflows/ecs.yml:36]:
```suggestion
        description: PHP versions to run action, separated by comma.
```

File: .github/workflows/ecs.yml [hunk 4]
```diff
@@ -0,0 +1,81 @@
...
+        required: true
+        type: string
+      tools:
```
Comment (vjik) [inline .github/workflows/ecs.yml:39]: It's need for CS fix?

File: docs/ecs.md [hunk 5]
```diff
@@ -0,0 +1,36 @@
...
+  push:
+    branches: ['main']
+    paths-ignore:
+      - 'docs/**'
+      - 'README.md'
+      - 'CHANGELOG.md'
+      - '.gitignore'
+      - '.gitattributes'
+      - 'infection.json.dist'
+      - 'phpunit.xml.dist'
```
Comment (vjik) [inline docs/ecs.md:24]: add psalm configuration

File: .github/workflows/ecs.yml [hunk 6]
```diff
@@ -0,0 +1,81 @@
...
+          tools: ${{ inputs.tools }}
+
+      - name: Update composer.
+        run: composer self-update ${{ inputs.composer-version }}
+
+      - name: Install dependencies with composer.
+        run: ${{ inputs.composer-command }}
+
+      - name: Run easy-coding-standard.
+        run: vendor/bin/ecs check src tests --ansi
```
Comment (vjik) [inline .github/workflows/ecs.yml:81]: Need autofix.

**Review by vjik — CHANGES_REQUESTED** [review]

File: .github/workflows/ecs.yml (hunk 2 above)
Comment (terabytesoftw) [reply .github/workflows/ecs.yml:32]: image

This is how it is in all CI.

File: .github/workflows/ecs.yml (hunk 3 above)
Comment (terabytesoftw) [reply .github/workflows/ecs.yml:36]: This is how it is in all CI.

File: .github/workflows/ecs.yml (hunk 6 above)
Comment (terabytesoftw) [reply .github/workflows/ecs.yml:81]: no.

File: docs/ecs.md (hunk 5 above)
Comment (terabytesoftw) [reply docs/ecs.md]: done.

File: .github/workflows/ecs.yml (hunk 1 above)
Comment (terabytesoftw) [reply .github/workflows/ecs.yml]: done.

File: .github/workflows/ecs.yml (hunk 4 above)
Comment (terabytesoftw) [reply .github/workflows/ecs.yml]: done.

File: .github/workflows/ecs.yml (hunk 2 above)
Comment (vjik) [reply .github/workflows/ecs.yml:22]: ECS is not test

File: .github/workflows/ecs.yml (hunk 6 above)
Comment (vjik) [reply .github/workflows/ecs.yml:65]: Yes) Make it optional via action configuration.

Comment (terabytesoftw) [reply .github/workflows/ecs.yml:65]: But we will have the same problem with the rector for commits, and i don't want to drag that here.

File: .github/workflows/ecs.yml (hunk 2 above)
Comment (terabytesoftw) [reply .github/workflows/ecs.yml:22]: I have no problem changing it, but then you have to adjust all the CI, so there is a standard.

File: .github/workflows/ecs.yml (hunk 6 above)
Comment (vjik) [reply .github/workflows/ecs.yml:65]: Add this optional and disabled by default

Comment (terabytesoftw) [reply .github/workflows/ecs.yml:65]: I prefer to do it in a separate PR, and try to solve the rector's problem once and for all.

File: .github/workflows/ecs.yml (hunk 2 above)
Comment (samdark) [reply .github/workflows/ecs.yml:22]: @vjik variant is more correct.

File: .github/workflows/ecs.yml (hunk 6 above)
Comment (samdark) [reply .github/workflows/ecs.yml:65]: We need auto-fix. That's why StyleCI fits us well. I don't want contributors to spend time on fixing code style issues at all.

**Review by vjik — APPROVED** [review]

**Pull request merged**